ocrd/ocrd/resolver.py

```diff
@@ -100,7 +105,46 @@ def download_to_directory(self, directory, url, basename=None, if_exists='skip',
             dst_path.write_bytes(src_path.read_bytes())
         else:
             log.debug("Downloading URL '%s' to '%s'", url, dst_path)
-            response = requests.get(url)
+            if 'OCRD_DOWNLOAD_RETRIES' in environ:
+                retries = retries or int(environ['OCRD_DOWNLOAD_RETRIES'])
+            retries = retries or 0
+            for _ in range(retries + 1):
+                try:
+                    if timeout is None and 'OCRD_DOWNLOAD_TIMEOUT' in environ:
+                        timeout = environ['OCRD_DOWNLOAD_TIMEOUT'].split(',')
+                        if len(timeout) > 1:
+                            timeout = tuple(float(x) for x in timeout)
+                        else:
+                            timeout = float(timeout[0])
+                    response = requests.get(url, timeout=timeout)
+                    if response.status_code in [
+                            # probably too wide:
+                            408, # Request Timeout
+                            409, # Conflict
+                            410, # Gone
+                            412, # Precondition Failed
+                            417, # Expectation Failed
+                            423, # Locked
+                            424, # Fail
+                            425, # Too Early
+                            426, # Upgrade Required
+                            428, # Precondition Required
+                            429, # Too Many Requests
+                            440, # Login Timeout
+                            500, # Internal Server Error
+                            503, # Service Unavailable
```

Comment:
Perhaps also `502 Bad Gateway` in case there is a connection issue between outward-facing and internal web service?

@MehmedGIT What HTTP status is returned when  retrieval of @subugoe images fails?

Reply:
I think everything from 4XX and 5XX should be included, except the 404.

@kba It's already covered there - 500.

Reply:
I don't think it's correct to include all error cases. For a retry mechanism we should focus on transient failures, not on permanent ones.

Reply:
404 and 410 are likely permanent, so should not be retried. But I would include all server side errors.